packages/typescript/src/components/FunctionArrowExpression.tsx

```diff
@@ -0,0 +1,92 @@
+import {
+  childrenArray,
+  findKeyedChild,
+  findUnkeyedChildren,
+  Scope,
+} from "@alloy-js/core";
+import { Children } from "@alloy-js/core/jsx-runtime";
+import { getCallSignatureProps } from "../utils.js";
+import { CallSignature, CallSignatureProps } from "./CallSignature.jsx";
+import { Declaration } from "./Declaration.jsx";
+import {
+  FunctionBody,
+  functionBodyTag,
+  FunctionParameters,
+  functionParametersTag,
+  FunctionTypeParameters,
+  functionTypeParametersTag,
+  getReturnType,
+} from "./FunctionBase.jsx";
+
+export interface FunctionArrowExpressionProps extends CallSignatureProps {
```

Comment:
ArrowFunctionExpression* I should think (ECMAScript grammar production)

Reply:
ES grammar doesn't have Expression actually, so fine leaving that out. Looks a lot better :)